src/test/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/TGContainerTest.groovy

```diff
@@ -177,4 +177,137 @@ class TGContainerTest extends ContainerTest {
             jenkins.assertLogContains(
                     "Report upload(s) unstable. Please see the logging of the uploads.", run)
     }
+
+    def "ttUploadReports: Test asynchronous upload using additionalSettings and verify with test.guide request"() {
+        given:
+            String script = """
+                node {
+                    withEnv(['ET_API_HOSTNAME=${etContainer.host}', 'ET_API_PORT=${etContainer.getMappedPort(ET_PORT)}','TG_HOSTNAME=${tgContainer.host}', 'TG_API_PORT=${tgContainer.getMappedPort(TG_PORT)}']) {
+                        def run_res = ttRunPackage 'test.pkg'
+    
+                        def upload_res = ttUploadReports testGuideUrl: 'http://${TG_ALIAS}:${TG_PORT}',
+                            credentialsId: 'authKey', 
+                            useSettingsFromServer: false,
+                            reportIds: [run_res.getReportId()],
+                            additionalSettings: [
+                                [name: "uploadAsync", value: "True"],
+                                [name: "setConstants", value: "BuildNumber=42;CustomConstant=Custom"],
+                                [name: "setAttributes", value: "CustomAttribute=Custom"]
+                            ]     
+                            
+                        sleep(2)
+                                                    
+                        def response = httpRequest (
+                                            ignoreSslErrors: true,
+                                            acceptType: 'APPLICATION_JSON',
+                                            contentType: 'APPLICATION_JSON',
+                                            httpMode: 'POST',
+                                            url: "http://\${TG_HOSTNAME}:\${TG_API_PORT}/api/report/testCaseExecutions/filter?projectId=1&offset=0&limit=10",
+                                            customHeaders: [
+                                                [name: 'TestGuide-AuthKey', value: "${TG_AUTH_KEY}"]
+                                            ],
+                                            requestBody: '''
+                                            {
+                                                "testCaseName": ["test"],
+                                                "atxIds": ["1"],
```

Comment:
The `atxId` is autogenerated and might not be one, for whatever reason

Just don't filter for the `atxId`